src/main/java/com/google/devtools/build/lib/skyframe/PackageArgsFunction.java

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.skyframe;
+
+import com.google.devtools.build.lib.cmdline.PackageIdentifier;
+import com.google.devtools.build.lib.cmdline.RepositoryName;
+import com.google.devtools.build.lib.events.Event;
+import com.google.devtools.build.lib.packages.LabelConverter;
+import com.google.devtools.build.lib.packages.PackageArgs;
+import com.google.devtools.build.lib.skyframe.RepoFileFunction.BadRepoFileException;
+import com.google.devtools.build.lib.skyframe.RepoFileFunction.RepoFileFunctionException;
+import com.google.devtools.build.lib.vfs.PathFragment;
+import com.google.devtools.build.skyframe.AbstractSkyKey;
+import com.google.devtools.build.skyframe.SkyFunction;
+import com.google.devtools.build.skyframe.SkyFunctionException;
+import com.google.devtools.build.skyframe.SkyFunctionName;
+import com.google.devtools.build.skyframe.SkyKey;
+import com.google.devtools.build.skyframe.SkyValue;
+import java.util.Map;
+import javax.annotation.Nullable;
+import net.starlark.java.eval.EvalException;
+
+/** A @{link SkyFunction} that returns the {@link PackageArgs} for a given repository. */
+public class PackageArgsFunction implements SkyFunction {
+  public static final PackageArgsFunction INSTANCE = new PackageArgsFunction();
+
+  /** {@link SkyValue} wrapping a PackageArgs. */
+  public static final class PackageArgsValue implements SkyValue {
+    public static final PackageArgsValue EMPTY = new PackageArgsValue(PackageArgs.EMPTY);
+
+    private final PackageArgs packageArgs;
+
+    public PackageArgsValue(PackageArgs packageArgs) {
+      this.packageArgs = packageArgs;
+    }
+
+    public PackageArgs getPackageArgs() {
+      return packageArgs;
+    }
+
+    @Override
+    public int hashCode() {
+      return packageArgs.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      if (other instanceof PackageArgsValue that) {
+        return that.packageArgs.equals(packageArgs);
+      } else {
+        return false;
+      }
+    }
+  }
+
+  public static Key key(RepositoryName repoName) {
+    return new Key(repoName);
+  }
+
+  /** Key type for {@link RepoFileValue}. */
```

Comment:
PackageArgsValue

Reply:
...or RepoPackageArgsValue to be compatible with your other suggestion. Done.